### PR TITLE
Upgrade gradle to 4.3.1

### PR DESCRIPTION
**Pull request by myott**

### CODE REVIEW

File: gradle/wrapper/gradle-wrapper.properties [hunk 1]
```diff
@@ -1,6 +1,5 @@
-#Thu Jun 16 09:15:57 CEST 2016
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-2.14-bin.zip
+distributionUrl=http\://services.gradle.org/distributions/gradle-4.3.1-bin.zip
```
Comment (drewcsillag) [inline gradle/wrapper/gradle-wrapper.properties]: better to keep the https here.

**Review by drewcsillag — CHANGES_REQUESTED** [review]

File: gradle/wrapper/gradle-wrapper.properties (hunk 1 above)
Comment (myott) [reply gradle/wrapper/gradle-wrapper.properties]: This is a weird problem with travis where it isn't able to pull it down. I encountered this in the process of open-sourcing commons and was able to resolve it. Can you send me the .travis.yml from there?

Comment (drewcsillag) [reply gradle/wrapper/gradle-wrapper.properties]:
```
language: java
script: ./gradlew clean build jacocoTestRep -i
jdk: oraclejdk8
env: TERM=dumb
after_success:
  - bash <(curl -s https://codecov.io/bash)
install: true
```

Comment (myott) [reply gradle/wrapper/gradle-wrapper.properties]: Oh, hmmm. I didn't by any chance just change https -> http in that repo, did I? I could have sworn I have to flip a few more things around.

**Comment by drewcsillag** [comment]
maybe compile with java7?

**Comment by drewcsillag** [comment]
java8 that is.

**Comment by myott** [comment]
Needed to use Precise... then I ran into https://github.com/travis-ci/travis-ci/issues/5227

Looking good now.

**Review by drewcsillag — APPROVED** [review]

**Pull request merged**